dev/buildtool/spin_commands.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implements SpinCLI support commands for buildtool."""
+
+import logging, os
+
+from collections import namedtuple
+
+from buildtool import (
+    BomSourceCodeManager,
+    RepositoryCommandFactory,
+    RepositoryCommandProcessor,
+
+    check_subprocess)
+
+from google.cloud import storage
+
+
+class DistArch(namedtuple('DistArch', ['dist', 'arch'])):
+  """Describes an distribution/architecture pair.
+  """
+  def __repl__(self):
+    return '{},{}'.format(self.dist, self.arch)
+
+
+DIST_ARCH_LIST = [
+  DistArch('darwin', 'amd64'),
+  DistArch('linux', 'amd64')
+]
+
+
+class BuildSpinCommand(RepositoryCommandProcessor):
+  def __init__(self, factory, options, **kwargs):
+    super(BuildSpinCommand, self).__init__(factory, options, **kwargs)
+    self.__gcs_uploader = SpinGcsUploader(options)
+
+  def _do_can_skip_repository(self, repository):
+    return repository.name != 'spin'
+
+  def _do_repository(self, repository):
+    """Implements RepositoryCommandProcessor interface."""
+    name = repository.name
+    build_version = self.scm.get_repository_service_build_version(repository)
+
+    if name == 'spin':
+      # TODO(jacobkiefer): go version && go env?
+      self.source_code_manager.ensure_local_repository(repository)
+      config_root = repository.git_dir
+
+      for dist_arch in DIST_ARCH_LIST:
+        # Sub-directory the binaries are stored in are specified by
+        # ${build_version}/${dist}.
+        version_bin_path = ('spin/{}/{}/{}/spin'
+                            .format(build_version, dist_arch.dist, dist_arch.arch))
+        nightly_bin_path = ('spin/nightly/{}/{}/spin'
+                            .format(dist_arch.dist, dist_arch.arch))
+
+        logging.info('Building spin binary for %s', dist_arch)
+        check_subprocess('env CGO_ENABLED=0 GOOS={} GOARCH={} go build .'
+                         .format(dist_arch.dist, dist_arch.arch),
+                         cwd=config_root)
+
+        spin_path = '{}/spin'.format(config_root)
+        self.__gcs_uploader.upload_from_filename(
+          version_bin_path, spin_path)
+        self.__gcs_uploader.upload_from_filename(
+          nightly_bin_path, spin_path)
+        os.remove(spin_path)
+
+      with open('latest', 'w') as latest_file:
```

Comment:
lastest_path = os.path.join(self.get_output_dir(), 'latest')

then just leave it in the output directory.